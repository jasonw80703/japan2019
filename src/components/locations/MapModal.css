#map {
  width: 100%;
}

.map-text {
  margin-top: 5px;
}