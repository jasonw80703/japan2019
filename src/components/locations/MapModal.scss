@import '../../assets/styles/colours';

#map {
  width: 100%;
}

.map-text {
  margin-top: 1%;
}

.recommendations-header {
  color: $light-red;
}