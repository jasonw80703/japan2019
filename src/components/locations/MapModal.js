import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-bootstrap/Modal';
import './MapModal.css';

export default class MapModal extends Component {
  render() {
    const {
      handleToggleModal,
      location,
      showModal,
      source,
      text,
    } = this.props;

    return (
      <Modal show={showModal} onHide={handleToggleModal} size='lg' className='map-modal'>
        <Modal.Header closeButton>
          <Modal.Title>{location}</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <div>
            <img src={source} className='map' alt='pic'/>
            {text && <p className='map-text'>{text}</p>}
          </div>
        </Modal.Body>
      </Modal>
    );
  }
}

MapModal.propTypes = {
  handleToggleModal: PropTypes.func.isRequired,
  location: PropTypes.string.isRequired,
  showModal: PropTypes.bool.isRequired,
  source: PropTypes.node.isRequired,
  text: PropTypes.string,
}