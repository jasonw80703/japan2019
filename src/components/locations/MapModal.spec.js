import React from 'react';
import EnzymeToJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import MapModal from './MapModal';

describe('<MapModal />', () => {
  const defaultProps = {
    handleToggleModal: () => {},
    location: 'San Francisco',
    showModal: true,
    source: '/static/media/sf11111.jpg',
  };

  it('renders as expected', () => {
    const tree = mount(<MapModal {...defaultProps} />)
    expect(EnzymeToJson(tree)).toMatchSnapshot();
  });
});