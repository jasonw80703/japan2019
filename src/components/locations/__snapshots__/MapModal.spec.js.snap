// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapModal /> renders as expected 1`] = `
<MapModal
  handleToggleModal={[Function]}
  location="San Francisco"
  showModal={true}
  source="/static/media/sf11111.jpg"
>
  <Bootstrap(Modal)
    className="map-modal"
    onHide={[Function]}
    show={true}
    size="lg"
  >
    <Modal
      animation={true}
      autoFocus={true}
      backdrop={true}
      bsPrefix="modal"
      className="map-modal"
      dialogAs={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "ModalDialog",
          "render": [Function],
        }
      }
      enforceFocus={true}
      keyboard={true}
      onHide={[Function]}
      restoreFocus={true}
      show={true}
      size="lg"
    >
      <Modal
        autoFocus={true}
        backdrop={true}
        backdropTransition={[Function]}
        className="map-modal modal"
        containerClassName="modal-open"
        enforceFocus={true}
        keyboard={true}
        manager={
          BootstrapModalManager {
            "adjustAndStore": [Function],
            "containers": Array [
              <body
                class="modal-open"
                style="padding-right: 0px;"
              >
                <div
                  class="fade modal-backdrop show"
                />
                <div
                  aria-modal="true"
                  class="fade map-modal modal show"
                  role="dialog"
                  style="display: block;"
                  tabindex="-1"
                >
                  <div
                    class="modal-dialog modal-lg"
                    role="document"
                  >
                    <div
                      class="modal-content"
                    >
                      <div
                        class="modal-header"
                      >
                        <div
                          class="modal-title h4"
                        >
                          San Francisco
                        </div>
                        <button
                          class="close"
                          type="button"
                        >
                          <span
                            aria-hidden="true"
                          >
                            ×
                          </span>
                          <span
                            class="sr-only"
                          >
                            Close
                          </span>
                        </button>
                      </div>
                      <div
                        class="modal-body"
                      >
                        <div>
                          <img
                            alt="pic"
                            class="map"
                            src="/static/media/sf11111.jpg"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </body>,
            ],
            "data": Array [
              Object {
                "classes": Array [
                  "modal-open",
                ],
                "modals": Array [
                  Object {
                    "add": [Function],
                    "backdrop": <div
                      class="fade modal-backdrop show"
                    />,
                    "dialog": <div
                      aria-modal="true"
                      class="fade map-modal modal show"
                      role="dialog"
                      style="display: block;"
                      tabindex="-1"
                    >
                      <div
                        class="modal-dialog modal-lg"
                        role="document"
                      >
                        <div
                          class="modal-content"
                        >
                          <div
                            class="modal-header"
                          >
                            <div
                              class="modal-title h4"
                            >
                              San Francisco
                            </div>
                            <button
                              class="close"
                              type="button"
                            >
                              <span
                                aria-hidden="true"
                              >
                                ×
                              </span>
                              <span
                                class="sr-only"
                              >
                                Close
                              </span>
                            </button>
                          </div>
                          <div
                            class="modal-body"
                          >
                            <div>
                              <img
                                alt="pic"
                                class="map"
                                src="/static/media/sf11111.jpg"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>,
                    "isTopModal": [Function],
                    "remove": [Function],
                    "setBackdropRef": [Function],
                    "setDialogRef": [Function],
                  },
                ],
                "overflowing": true,
                "style": Object {
                  "overflow": "",
                  "paddingRight": "",
                },
              },
            ],
            "handleContainerOverflow": true,
            "hideSiblingNodes": true,
            "modals": Array [
              Object {
                "add": [Function],
                "backdrop": <div
                  class="fade modal-backdrop show"
                />,
                "dialog": <div
                  aria-modal="true"
                  class="fade map-modal modal show"
                  role="dialog"
                  style="display: block;"
                  tabindex="-1"
                >
                  <div
                    class="modal-dialog modal-lg"
                    role="document"
                  >
                    <div
                      class="modal-content"
                    >
                      <div
                        class="modal-header"
                      >
                        <div
                          class="modal-title h4"
                        >
                          San Francisco
                        </div>
                        <button
                          class="close"
                          type="button"
                        >
                          <span
                            aria-hidden="true"
                          >
                            ×
                          </span>
                          <span
                            class="sr-only"
                          >
                            Close
                          </span>
                        </button>
                      </div>
                      <div
                        class="modal-body"
                      >
                        <div>
                          <img
                            alt="pic"
                            class="map"
                            src="/static/media/sf11111.jpg"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>,
                "isTopModal": [Function],
                "remove": [Function],
                "setBackdropRef": [Function],
                "setDialogRef": [Function],
              },
            ],
            "restore": [Function],
            "scrollbarSize": 0,
          }
        }
        onClick={[Function]}
        onEnter={[Function]}
        onEntering={[Function]}
        onExited={[Function]}
        onHide={[Function]}
        onMouseUp={[Function]}
        renderBackdrop={[Function]}
        restoreFocus={true}
        show={true}
        style={
          Object {
            "paddingLeft": undefined,
            "paddingRight": undefined,
          }
        }
        transition={[Function]}
      >
        <Portal
          containerInfo={
            <body
              class="modal-open"
              style="padding-right: 0px;"
            >
              <div
                class="fade modal-backdrop show"
              />
              <div
                aria-modal="true"
                class="fade map-modal modal show"
                role="dialog"
                style="display: block;"
                tabindex="-1"
              >
                <div
                  class="modal-dialog modal-lg"
                  role="document"
                >
                  <div
                    class="modal-content"
                  >
                    <div
                      class="modal-header"
                    >
                      <div
                        class="modal-title h4"
                      >
                        San Francisco
                      </div>
                      <button
                        class="close"
                        type="button"
                      >
                        <span
                          aria-hidden="true"
                        >
                          ×
                        </span>
                        <span
                          class="sr-only"
                        >
                          Close
                        </span>
                      </button>
                    </div>
                    <div
                      class="modal-body"
                    >
                      <div>
                        <img
                          alt="pic"
                          class="map"
                          src="/static/media/sf11111.jpg"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </body>
          }
        >
          <BackdropTransition
            appear={true}
            in={true}
          >
            <Fade
              appear={true}
              in={true}
              mountOnEnter={false}
              timeout={300}
              unmountOnExit={false}
            >
              <Transition
                addEndListener={[Function]}
                appear={true}
                enter={true}
                exit={true}
                in={true}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={300}
                unmountOnExit={false}
              >
                <div
                  className="fade modal-backdrop show"
                  onClick={[Function]}
                />
              </Transition>
            </Fade>
          </BackdropTransition>
          <DialogTransition
            appear={true}
            in={true}
            onEnter={[Function]}
            onEntering={[Function]}
            onExited={[Function]}
            unmountOnExit={true}
          >
            <Fade
              appear={true}
              in={true}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntering={[Function]}
              onExited={[Function]}
              timeout={300}
              unmountOnExit={true}
            >
              <Transition
                addEndListener={[Function]}
                appear={true}
                enter={true}
                exit={true}
                in={true}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={300}
                unmountOnExit={true}
              >
                <div
                  aria-modal={true}
                  className="fade map-modal modal show"
                  onClick={[Function]}
                  onMouseUp={[Function]}
                  role="dialog"
                  style={
                    Object {
                      "paddingLeft": undefined,
                      "paddingRight": undefined,
                    }
                  }
                  tabIndex={-1}
                >
                  <ModalDialog
                    onMouseDown={[Function]}
                    role="document"
                    size="lg"
                  >
                    <div
                      className="modal-dialog modal-lg"
                      onMouseDown={[Function]}
                      role="document"
                    >
                      <div
                        className="modal-content"
                      >
                        <ModalHeader
                          closeButton={true}
                          closeLabel="Close"
                        >
                          <div
                            className="modal-header"
                          >
                            <ModalTitle>
                              <ForwardRef
                                className="modal-title"
                              >
                                <div
                                  className="modal-title h4"
                                >
                                  San Francisco
                                </div>
                              </ForwardRef>
                            </ModalTitle>
                            <CloseButton
                              label="Close"
                              onClick={[Function]}
                            >
                              <button
                                className="close"
                                onClick={[Function]}
                                type="button"
                              >
                                <span
                                  aria-hidden="true"
                                >
                                  ×
                                </span>
                                <span
                                  className="sr-only"
                                >
                                  Close
                                </span>
                              </button>
                            </CloseButton>
                          </div>
                        </ModalHeader>
                        <ModalBody>
                          <div
                            className="modal-body"
                          >
                            <div>
                              <img
                                alt="pic"
                                className="map"
                                src="/static/media/sf11111.jpg"
                              />
                            </div>
                          </div>
                        </ModalBody>
                      </div>
                    </div>
                  </ModalDialog>
                </div>
              </Transition>
            </Fade>
          </DialogTransition>
        </Portal>
      </Modal>
    </Modal>
  </Bootstrap(Modal)>
</MapModal>
`;
